### PR TITLE
Feature/gridedit 577 add search distance

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: libs/MeshKernel/include/MeshKernel/Contacts.hpp [hunk 1]
```diff
@@ -151,6 +152,9 @@ namespace meshkernel
         /// @return True if the contact is crossing an existing contact
         [[nodiscard]] bool IsContactIntersectingContact(size_t node, size_t face) const;
 
+        void Connect1dNodesWithCrossingFaces(size_t node,
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/include/MeshKernel/Contacts.hpp:155]: missing doxy

Comment (ahmad-el-sayed) [inline libs/MeshKernel/include/MeshKernel/Contacts.hpp:155]: Missing doxy

File: libs/MeshKernel/src/Contacts.cpp [hunk 2]
```diff
@@ -53,7 +54,29 @@ void Contacts::ComputeSingleContacts(const std::vector<bool>& oneDNodeMask,
         {
             m_mesh1dIndices.emplace_back(n);
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/src/Contacts.cpp:55]: push_back?

File: libs/MeshKernel/include/MeshKernel/Contacts.hpp [hunk 3]
```diff
@@ -151,6 +152,9 @@ namespace meshkernel
...
+                                             double distanceFactor);
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/include/MeshKernel/Contacts.hpp:156]: rename to `projectionFactor` for consistency?

File: libs/MeshKernel/src/Contacts.cpp [hunk 4]
```diff
@@ -53,7 +54,29 @@ void Contacts::ComputeSingleContacts(const std::vector<bool>& oneDNodeMask,
...
+{
+    const auto projectedNode = m_mesh1d->ComputeProjectedNode(node, distanceFactor);
+
+    const auto [intersectedFace, intersectedEdge] = m_mesh2d->IsSegmentCrossingABoundaryEdge(m_mesh1d->m_nodes[node], projectedNode);
+    if (intersectedFace != constants::missing::sizetValue &&
+        intersectedEdge != constants::missing::sizetValue &&
+        !IsContactIntersectingMesh1d(node, intersectedFace) &&
+        !IsContactIntersectingContact(node, intersectedFace))
+    {
+        m_mesh1dIndices.emplace_back(node);
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/src/Contacts.cpp:78]: push_back?

File: libs/MeshKernel/src/Contacts.cpp (hunk 2 above)
Comment (lucacarniato) [reply libs/MeshKernel/src/Contacts.cpp:55]: should not make a difference for named variables

File: libs/MeshKernel/src/Contacts.cpp (hunk 4 above)
Comment (lucacarniato) [reply libs/MeshKernel/src/Contacts.cpp:78]: as above

**Pull request merged**